conans/client/command.py

```diff
@@ -58,6 +58,11 @@ def __call__(self, parser, namespace, values, option_strings=None):  # @UnusedVa
                 dest.extend(values)
             except ValueError:
                 dest.append(values)
+        # --build --build=foo == ["*", "foo"]
+        elif hasattr(namespace, "build") and isinstance(namespace.build, list):
+            if len(namespace.build) == 0 or \
+               any(str(it).startswith("!") for it in namespace.build):
+                dest.append("*")
```

Comment:
Having a exclude pattern shouldn't be tied to building from source everything. You could have ``--build=mypkgs* --build=!mypkgs_special`` and it should build all packages starting with ``mypkgs`` but not the excluded one.

Please try to avoid as much as possible to particularize a generic thing as ``Extender`` with custom logic from one of its consumer, it is an anti-pattern and should be used only in case of high necessity.

Reply:
I also think this might still be changing behavior, and this will probably bring issues, better to avoid it.

Reply:
By default, when we pass `--build --build=!foo`, our parse only returns `!foo`, so doesn't matters whether `--build` is listed as command argument or not. This new conditional, only accepts `--build=!` when `--build` is present too. Or, we can create an implicit rule, like if you passed only `--build=!foo`, it means build all from sources, except for foo.

Reply:
I would prefer it to be explicit, and move in the direction of Conan 2.0 CLI than adding it magically. So we could document that ``--build=!xxx`` requires other patterns to be positively defined, even if ``--build=*``. Better than customize the ``Extender`` class with specific attribute logic (in other command --build could mean a different thing (actually it does in ``conan build --build`` command, and this is a risk of failure.

Reply:
> -build=!xxx requires other patterns to be positively defined

by contrary, we don't need to define an extra argument, without this workaround. If you agree, I'll remove it and keep only `--build=!xxx`, which is enough for us.